polkadot/collator/src/lib.rs

```diff
@@ -45,25 +45,40 @@
 //! to be performed, as the collation logic itself.
 
 extern crate futures;
+extern crate substrate_client as client;
 extern crate substrate_codec as codec;
 extern crate substrate_primitives as primitives;
+extern crate ed25519;
+
+extern crate polkadot_api;
+extern crate polkadot_cli;
 extern crate polkadot_runtime;
 extern crate polkadot_primitives;
 
+#[macro_use]
+extern crate log;
+
 use std::collections::{BTreeSet, BTreeMap};
+use std::sync::Arc;
 
-use futures::{stream, Stream, Future, IntoFuture};
-use polkadot_primitives::parachain::{self, CandidateSignature, ConsolidatedIngress, Message, Id as ParaId};
+use futures::{future, stream, Stream, Future, IntoFuture};
+use client::BlockchainEvents;
+use polkadot_api::PolkadotApi;
+use polkadot_primitives::BlockId;
+use polkadot_primitives::parachain::{self, BlockData, HeadData, ConsolidatedIngress, Collation, Message, Id as ParaId};
+use polkadot_cli::{ClientError, ServiceComponents, ClientBackend, PolkadotBlock, StateMachineBackend, Service};
+use polkadot_cli::Application;
 
 /// Parachain context needed for collation.
 ///
 /// This can be implemented through an externally attached service or a stub.
-pub trait ParachainContext {
+pub trait ParachainContext: Clone {
```

Comment:
Should we add something like this to the comment?

`This is expected to be a lightweight, shared type like an Arc.`

Reply:
yeah, I will add it